packages/fuchsia_remote_debug_protocol/lib/src/common/network.dart

```diff
@@ -0,0 +1,34 @@
+// Copyright 2018 The Flutter Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+import 'dart:core';
+
+/// Determines whether the address is valid IPv6 or IPv4 format.
+///
+/// Throws an `ArgumentError` if the address is neither.
+void validateAddress(String address) {
+  if (!(isIpV4Address(address) || isIpV6Address(address))) {
+    throw new ArgumentError('"$address" is neither valid IPv4 nor IPv6');
+  }
+}
+
+/// Returns true if the address is a valid IPv6 address.
```

Comment:
nit: for consistency, we should probably stick with one of "has format" or "is address".

Reply:
Done.